Sitepipe doesn't build against stack resolver lts-18.16

```
/tmp/stack-a1cc98b9d6004bc4/sitepipe-0.4.0.1/src/SitePipe/Readers.hs:52:30: error:
    • Couldn't match type ‘Control.Monad.Trans.State.Lazy.StateT
                             CommonState IO’
                     with ‘StateT s0 IO’
      NB: ‘StateT’ is defined in ‘Control.Monad.Trans.State.Strict’
          ‘Control.Monad.Trans.State.Lazy.StateT’
            is defined in ‘Control.Monad.Trans.State.Lazy’
      Expected type: StateT s0 IO (Either PandocError a)
        Actual type: Control.Monad.Trans.State.Lazy.StateT
                       CommonState IO (Either PandocError a)
    • In the second argument of ‘($)’, namely
        ‘runExceptT $ unPandocIO m’
      In a stmt of a 'do' block:
        z <- flip evalStateT def $ runExceptT $ unPandocIO m
      In the expression:
        do z <- flip evalStateT def $ runExceptT $ unPandocIO m
           case z of
             Left e -> throwM e
             Right a -> pure a
   |
52 |   z <- flip evalStateT def $ runExceptT $ unPandocIO m
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^
```

```
$ stack ls dependencies

Cabal 3.2.1.0
Glob 0.10.1
HTTP 4000.3.16
HsYAML 0.2.1.0
JuicyPixels 3.3.6
MissingH 1.4.3.0
QuickCheck 2.14.2
SHA 1.6.4.4
StateVar 1.2.2
adjunctions 4.4
aeson 1.5.6.0
aeson-pretty 0.8.9
ansi-terminal 0.11
ansi-wl-pprint 0.6.9
array 0.5.4.0
asn1-encoding 0.9.6
asn1-parse 0.9.5
asn1-types 0.3.4
assoc 1.0.2
async 2.2.4
attoparsec 0.13.2.5
base 4.14.3.0
base-compat 0.11.2
base-compat-batteries 0.11.2
base-orphans 0.8.6
base16-bytestring 1.0.2.0
base64-bytestring 1.1.0.0
basement 0.0.12
bifunctors 5.5.11
binary 0.8.8.0
blaze-builder 0.4.2.2
blaze-html 0.9.1.2
blaze-markup 0.8.2.8
bytestring 0.10.12.0
cabal-doctest 1.0.8
call-stack 0.3.0
case-insensitive 1.2.1.0
cereal 0.5.8.2
citeproc 0.4.0.1
cmdargs 0.10.21
cofree-coffee-cast 0.1.0.0
colour 2.3.6
commonmark 0.2.1.1
commonmark-extensions 0.2.2.1
commonmark-pandoc 0.2.1.1
comonad 5.0.8
conduit 1.3.4.2
conduit-extra 1.3.5
connection 0.3.1
constraints 0.13.1
containers 0.6.5.1
contravariant 1.5.5
cookie 0.4.5
cryptonite 0.29
data-default 0.7.1.1
data-default-class 0.1.2.0
data-default-instances-containers 0.0.1
data-default-instances-dlist 0.0.1
data-default-instances-old-locale 0.0.1
data-fix 0.3.2
deepseq 1.4.4.0
digest 0.0.1.3
directory 1.3.6.0
distributive 0.6.2.1
dlist 1.0
doclayout 0.3.1.1
doctemplates 0.9
either 5.0.1.1
emojis 0.1.2
enclosed-exceptions 1.0.3
errors 2.3.0
exceptions 0.10.4
file-embed 0.0.15.0
filepath 1.4.2.1
free 5.1.7
ghc-boot-th 8.10.7
ghc-prim 0.6.1
haddock-library 1.10.0
hashable 1.3.0.0
hourglass 0.2.12
hslogger 1.3.1.0
hslua 1.3.0.2
hslua-module-path 0.1.0.1
hslua-module-system 0.2.2.1
hslua-module-text 0.3.0.1
http-client 0.6.4.1
http-client-tls 0.3.5.3
http-types 0.12.3
indexed-traversable 0.1.2
integer-gmp 1.0.3.0
integer-logarithms 1.0.3.1
invariant 0.5.5
ipynb 0.1.0.2
jira-wiki-markup 1.4.0
kan-extensions 5.2.3
lens 4.19.2
lens-aeson 1.1.2
libyaml 0.1.2
lifted-async 0.10.2.2
lifted-base 0.2.3.12
megaparsec 9.0.1
memory 0.15.0
mime-types 0.1.0.9
monad-control 1.0.3.1
mono-traversable 1.0.15.3
mtl 2.2.2
mustache 2.3.1
network 3.1.1.1
network-bsd 2.8.1.0
network-uri 2.6.4.1
old-locale 1.0.0.7
old-time 1.1.0.3
optparse-applicative 0.16.1.0
pandoc 2.14.0.3
pandoc-types 1.22.1
parallel 3.2.2.0
parsec 3.1.14.0
parser-combinators 1.2.1
pem 0.2.4
pretty 1.1.3.6
primitive 0.7.3.0
process 1.6.13.2
profunctors 5.6.2
random 1.2.0
reflection 2.1.6
regex-base 0.94.0.1
regex-compat 0.95.2.1
regex-posix 0.96.0.1
resourcet 1.2.4.3
rts 1.0.1
safe 0.3.19
scientific 0.3.7.0
semigroupoids 5.3.6
semigroups 0.19.2
shelly 1.9.0
sitepipe 0.4.0.1
skylighting 0.10.5.2
skylighting-core 0.10.5.2
socks 0.6.1
split 0.2.3.4
splitmix 0.1.0.4
stm 2.5.0.1
streaming-commons 0.2.2.1
strict 0.4.0.1
syb 0.7.2.1
tagged 0.8.6.1
tagsoup 0.14.8
template-haskell 2.16.0.0
temporary 1.3
texmath 0.12.3.2
text 1.2.4.1
text-conversions 0.3.1
th-abstraction 0.4.3.0
th-compat 0.1.3
th-lift 0.8.2
th-lift-instances 0.1.18
these 1.1.1.1
time 1.9.3
time-compat 1.9.5
tls 1.5.5
transformers 0.5.6.2
transformers-base 0.4.6
transformers-compat 0.6.6
type-equality 1
typed-process 0.2.7.0
unicode-collation 0.1.3
unicode-transforms 0.3.7.1
uniplate 1.6.13
unix 2.7.2.2
unix-compat 0.5.3
unliftio-core 0.2.0.1
unordered-containers 0.2.14.0
utf8-string 1.0.2
uuid-types 1.0.5
vector 0.12.3.1
vector-algorithms 0.8.0.4
void 0.7.3
x509 1.7.5
x509-store 1.6.7
x509-system 1.6.6
x509-validation 1.6.11
xml 1.3.14
xml-conduit 1.9.1.1
xml-types 0.3.8
yaml 0.11.7.0
zip-archive 0.4.1
zlib 0.6.2.3
```

--- a/sitepipe.cabal
+++ b/sitepipe.cabal
@@ -1,13 +1,13 @@
 cabal-version: 1.12
 
--- This file has been generated from package.yaml by hpack version 0.31.2.
+-- This file has been generated from package.yaml by hpack version 0.34.4.
 --
 -- see: https://github.com/sol/hpack
 --
--- hash: 3616475eec5c75349ec356fdb80cf766bb5e195fb279b6e02fad093ae9abe840
+-- hash: 7e40b9776705eedf82dd97e4999739013a344bbeb230c00a75b169c1f995a501
 
 name:           sitepipe
-version:        0.4.0.0
+version:        0.4.0.1
 synopsis:       A simple to understand static site generator
 category:       Web
 homepage:       https://github.com/ChrisPenner/sitepipe#readme
@@ -17,7 +17,8 @@ maintainer:     christopher.penner@gmail.com
 copyright:      2017 Chris Penner
 license:        BSD3
 license-file:   LICENSE
-tested-with:    GHC==8.0.2 GHC==8.2.1
+tested-with:
+    GHC==8.0.2 GHC==8.2.1
 build-type:     Simple
 extra-source-files:
     README.md

--- a/src/SitePipe/Readers.hs
+++ b/src/SitePipe/Readers.hs
@@ -16,7 +16,7 @@ module SitePipe.Readers
   ) where
 
 import Control.Monad.Trans.Except
-import Control.Monad.Trans.State.Strict
+import Control.Monad.Trans.State
 import Control.Monad.Catch
 import Text.Pandoc
 import Text.Pandoc.Options

--- a/stack.yaml
+++ b/stack.yaml
@@ -15,7 +15,7 @@
 # resolver:
 #  name: custom-snapshot
 #  location: "./custom-snapshot.yaml"
-resolver: lts-14.1
+resolver: lts-18.16
 
 # User packages to be built.
 # Various formats can be used as shown in the example below.
@@ -40,7 +40,8 @@ packages:
 # - '../mustache'
 # Dependency packages to be pulled from upstream that are not in the resolver
 # (e.g., acme-missiles-0.3)
-extra-deps: []
+extra-deps:
+- MissingH-1.5.0.1@sha256:f662d35e54eee556bf6f1a8ec4319728985432689e844e611b02e958137b51a2,4651
 
 # Override default flag values for local packages and extra-deps
 flags: {}

--- a/stack.yaml.lock
+++ b/stack.yaml.lock
@@ -3,10 +3,17 @@
 # For more information, please see the documentation at:
 #   https://docs.haskellstack.org/en/stable/lock_files
 
-packages: []
+packages:
+- completed:
+    hackage: MissingH-1.5.0.1@sha256:f662d35e54eee556bf6f1a8ec4319728985432689e844e611b02e958137b51a2,4651
+    pantry-tree:
+      size: 5272
+      sha256: d9e6962a5bda62d0c1e6f8c3536b1f43d39eb3456e478c2f1e2a9f50bbb13308
+  original:
+    hackage: MissingH-1.5.0.1@sha256:f662d35e54eee556bf6f1a8ec4319728985432689e844e611b02e958137b51a2,4651
 snapshots:
 - completed:
-    size: 523448
-    url: https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/lts/14/1.yaml
-    sha256: 0045b9bae36c3bb2dd374c29b586389845af1557eea0423958d152fc500d4fbf
-  original: lts-14.1
+    size: 586286
+    url: https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/lts/18/16.yaml
+    sha256: cdead65fca0323144b346c94286186f4969bf85594d649c49c7557295675d8a5
+  original: lts-18.16

